### DISCUSSION TITLE
No "official" release outside github?

### DISCUSSION THREAD

**Discussion by augsf** [goldendict/goldendict#1594] [Q&A] [answered]
Hi all, sorry if this is a dumb question, but it seems that many users of GD compile their own install from the code available here on github. Isn't there, like it is with VLC or Handbrake, an "official" packaged release of GD? I use the flatpak version available on flathub because the dev who packaged it seems to be legit, but by looking at the discussions in the issues section it seems that there's been plenty of trouble with less-than-legit forks being peddled out there.

Anyhow, thank you all for the good work!

**Comment by darnn** [comment]
There's this:
https://github.com/xiaoyifang/goldendict/releases/

Although last I checked (a while ago, granted), it was much less stable and usable than the less-than-legit fork you're talking about, which at the very least fixes this:
https://github.com/goldendict/goldendict/issues/1374#issuecomment-848963368

**Reply by shenlebantongying** [reply]
Can you explain further what do you mean by "less stable and usable"? Some usability ideas could be useful.

**Comment by darnn** [comment]
Actually, it has improved a lot in the past few months. I would say right now my main complaint is this: When displaying a website, if "as link" is enabled, it shows a second scrollbar for the website, inside the area where the results are rendered, which already has its own scrollbar:
![gd22-apple1](https://user-images.githubusercontent.com/19672444/209099437-73633807-39ec-4acd-bb9f-274b2662c688.png)
Aside from being undesirable in its own right, it creates issues when trying to navigate using the keyboard. For instance, since the "as link" website steals the focus, for lack of a better term, if you try to use pgdn to move down in the results, it will first scroll through the entire first website and only then scroll through the actual results window.

If "as link" is disabled, it creates an unnecessary horizontal scrollbar, and fails to display the website correctly:
![gd22-apple2](https://user-images.githubusercontent.com/19672444/209100043-bfe6a849-270f-4dda-b365-6a37eb99bf33.png)

In both situations, moving the (vertical) scroll bar with the mouse doesn't work like it normally would in any other Windows program.

**Reply by shenlebantongying** [reply]
Thanks for your feedback. I think the problems you mentioned are solvable.

The scrollbar exists because it is a <iframe> with its content height undecided before creation.

There should be a solution, but I am not an expert on the web in general. Someone with good knowledge of the frontend should know a proper solution.

Also, I think the "As link" functionality should be removed since modern websites all have tons of JS & CS. Blending websites together will for sure cause visual bugs.

**Reply by xiaoyifang** [reply]
> The scrollbar exists because it is a <iframe> with its content height undecided before creation.

this is intended ,as some website's height is too large.

remove the following parameters can enable the website to its full height.

![image](https://user-images.githubusercontent.com/105986/209112633-2de45f24-6cfc-4676-95eb-b3be93374b5c.png)

**Comment by darnn** [comment]
Well, like, that's the thing, "as link" works fine in GD++. When it's turned off the same issues appear. But it does show it's possible to resolve these things (I am also not an expert, but presumably rendering the entire page first, including the websites, and only then displaying it, which I think is what it does as it is). Also, it would be great to have a uBlock(/AdBlock etc.) style element picker to block unnecessary parts of the webpage so that only the actual part wit the definition remains. It's possible to add them to the style.css file manually, but that's tedious, and also, not all of it actually works the way it should, not all the elements get blocked. And even then, it only works when "as link" is disabled, so I essentially can't use it at all.

Edit, in response to the comment I hadn't seen before posting: Ideally, the height would get resized _and_ there wouldn't be a second scrollbar (once again, as is the case in GD++).

**Comment by augsf** [comment]
Guys, this is all fascinating, but wouldn't it belong in the issues section of xiaoyifang's fork?

In other words, the question I raised is this: From where should a regular, non-dev user who wants to use GoldenDict download it?

I was a windows user before and for some time used a version from the Microsoft Store that turned out to be spyware #1319 Now I'm using Ubuntu with flatpak and downloaded @proletarius101 's version from Flathub because he seems to have contributed to the main project, but I don't know if his is actually just a repackaging or actually a new fork. As for @xiaoyifang 's fork, I'm sure he is putting some very good work into it, and he seems to be on good terms with the devs of the original project, but perhaps I'm a little biased against AppImages? And even if not, a user has to come here to GitHub and choose from several options that use different Qt versions. Not exactly noob-friendly.

GoldenDict is a very useful piece of software, and Linux desktop, of all platforms, deserve some good software like that, not only for devs, but for non-dev users as well. So maybe you guys could join forces to produce one or two official packages and put it out there. Pls don't hate on me. This is just a suggestion.

**Reply by shenlebantongying** [reply]
It will be in the official repo of Debian 12 & Ubuntu 23.04

https://packages.debian.org/bookworm/goldendict-webengine

**Reply by proletarius101** [reply]
Not a fork. No substantial changes.

**Reply by augsf** [reply]
Thanks @shenlebantongying and @proletarius101 for the clarifications.

Seems, however, that we will have two different versions, under the same name, one a .deb the other a flatpak?

**Comment by darnn** [comment]
For what it's worth, I absolutely agree that there should be an official release of a new version, or at least a new early access build, like the ones that used to go up here:
https://github.com/goldendict/goldendict/wiki/Early-Access-Builds-for-Windows

**Comment by nonwill** [comment]
> It will be in the official repo of Debian 12 & Ubuntu 23.04
> 
> https://packages.debian.org/bookworm/goldendict-webengine



For what it's worth, the quality of @vedgy ‘s pr/implementation is 1000 orders of magnitude higher than that of xiaoyifang. So, some discussions are just too much nonsense to say, and not too hard to hear.
It's unfortunate that No One Here chose the implementation of @vedgy. It's comical for Debian to have someone who can't even use pointers well preside over this important thing.

**Reply by shenlebantongying** [reply]
Debian is certainly not a friend of someone privatizing public code. Sorry to hear that you have no friends in the free software community.

**Reply by nonwill** [reply]
I'm not sad at all, and you should not be too sorry too. And I just believe that the self-proclaimed gods and goddesses are just that. hahahaha.

**Comment by augsf** [comment]
What's comical is toxicity pushed to its farcical limits. But hey, this is what we have to put up with.

**Comment by nonwill** [comment]
It is just a comical excuse for the incompetent, and it is hard to change the nature by changeing _**names**_.

**Reply by augsf** [reply]
I hoped I'd made it clear that I was not particularly interested in your input, but you're obviously not the type to take a hint. Maybe you reckon you'll reduce competition to your warez by trying to demoralize devs working on open-source GD with taunts and disruptions here. It won't work.

**Reply by nonwill** [reply]
**_No expectations at all as you said_**.  And as your expectations, seems that you also don't need to take prompts from others because whom, those **_bugers_** can't represent the level of GD and Debian communities.

**Comment by goyalyashpal** [comment]
i am also looking for a FOSS dictionary for windows, and none but golden dict exists. And there's a LOT of devt being going on (as seen from the commit history) - would love to see an official release...

or maybe some list of blockers which are blocking the release would be helpful too so that someone interested can help to realise the stable build.

the ongoing devt can and will ofcourse continue, but having one stable release will bring things to their much needed state.

**Comment by vedgy** [comment, marked as answer]
This discussion mostly duplicates issues #839 and #953. There isn't much hope for a new release, complete with fresh builds for all platforms, anytime soon. Fresh Windows executables are being made. GNU/Linux users are mostly fine too, plus I plan to tag a new date-based version soon. But GoldenDict lacks macOS contributions. A CI setup contribution could allow to produce a macOS build at each commit to master.

**Reply by goyalyashpal** [reply]
> There isn't much hope for a new release, complete with fresh builds for all platforms, anytime soon. ... lacks macos

inkscape had similar issue with version 1.0 or 1.1 and they released the macos as beta for that version with a notice in release logs clearly mentioning this hurdle.

**Reply by augsf** [reply]
Well, not really duplicates. What I pointed out is that there are no user-friendly distribution end points where (non-dev) users can rest assured that the app they're getting is coming directly from the developers and, as pointed out by @vedgy himself, no clear version numbering (yet). It is also not a matter of Goldendict being available on all platforms all at the same time. If one platform lacks contributors there is no reason for the other platforms to suffer.

**Reply by vedgy** [reply]
> Well, not really duplicates.

#953 asks for a release as well. I think neither @Abs62 nor I can make changes to the website http://goldendict.org. The more recent discussion #1620 also asks for the website and forum updates. So the issue discussed here is kind of covered in other places.

No GoldenDict developer is willing to maintain a Flatpak/AppImage/Snap version. You are lucky that a Flatpak enthusiast has decided to maintain a GoldenDict package on Flathub :)

Maybe we'll have a clearly defined Windows build corresponding to the planned date-based tag. Depends on @Abs62's decision probably. And maybe an AppImage as well. Looks like [AppImage CI updates](https://github.com/Abs62/goldendict/releases/tag/continuous) stopped after January 19 for some reason.

**Reply by vedgy** [reply]
In case you don't know, Flatpak/AppImage/Snap versions of software often suffer from packaging-specific problems, e.g. system theme integration, so maintaining them is not trivial. Thankfully, building and installing GoldenDict on a GNU/Linux system should be straightforward and fast, once you figure out which dependencies need to be installed. Users could share distro-specific scripts that build GoldenDict from sources. Some distributions have user package repositories, such as AUR. For the distros that don't have such mechanisms, we could store the build scripts/user packages in this repository and link to them from the README.

**Reply by goyalyashpal** [reply]
> neither @Abs62 nor I can make changes to the website http://goldendict.org. The more recent discussion #1620 also asks for the website and forum updates

well, it doesnt need to be on site. having a release on github with "stable" label is enough as well

**Reply by vedgy** [reply]
> Looks like [AppImage CI updates](https://github.com/Abs62/goldendict/releases/tag/continuous) stopped after January 19 for some reason.

Logged [an issue](https://github.com/Abs62/goldendict/issues/3) about this. Turns out Travis CI is now paid-only. Whoever cares about AppImage can migrate to a gratis CI service.

**Reply by goyalyashpal** [reply]
wow, the release was just 18-12 days away. good work @ vedgy and everyone else involved :)

https://github.com/goldendict/goldendict/releases/tag/1.5.0

**Reply by lonix1** [reply]
Hi @vedgy I was very excited to learn that this has been updated (I'm using a version that's probably 10+ years old).

But I'm on linux and I noticed the releases page only has windows binaries. Are there linux x64 files too?

**Reply by vedgy** [reply]
> Are there linux x64 files too?

No. GNU/Linux users are supposed to either use GoldenDict packages provided by their distributions or build from sources.

An AppImage used to be built at each GoldenDict commit, but the CI fell into disrepair: https://github.com/Abs62/goldendict/issues/3 . AppImage has its own downsides, e.g. poor integration with the system.

There is a GoldenDict flatpak maintained by an infrequent GoldenDict contributor. So that's another alternative, but you'd have to report flatpak issues at the GoldenDict flatpak repository, not here.

**Comment by vedgy** [comment]
If anyone is interested in a GoldenDict AppImage, please test https://github.com/goldendict/goldendict/issues/905#issuecomment-3213090949 so we can merge the AppImage CI pull request with more confidence.
